packages/flutter_tools/test/general.shard/features_config_test.dart

```diff
@@ -0,0 +1,169 @@
+  }
+
+  test('returns null if cannot be overriden', () {
+    expect(
+      isEnabled(
+        _noConfigFeature,
+        environment: <String, String>{'ENABLE_FLAG': 'true'},
+        globalConfig: <String, Object>{'enable-flag': true},
+        projectManifest: '''
+        flutter:
+          config:
+            enable-flag: true
+        ''',
+      ),
+      isNull,
+    );
+  });
+
+  test('returns null if every source is omitted', () {
+    expect(isEnabled(_configAndEnvFeature), isNull);
+  });
+
+  test('overrides from local manifest', () {
+    expect(
+      isEnabled(
+        _configOnlyFeature,
+        projectManifest: '''
+        flutter:
+          config:
+            enable-flag: true
+        ''',
+      ),
+      isTrue,
+    );
+  });
+
+  test('local manifest config must be a map', () {
+    expect(
+      () => isEnabled(
+        _configOnlyFeature,
+        projectManifest: '''
+        flutter:
+          config: true
+        ''',
+      ),
+      throwsToolExit(message: 'must be a map'),
+    );
+  });
+
+  test('local manifest value must be a boolean', () {
+    expect(
+      () => isEnabled(
+        _configOnlyFeature,
+        projectManifest: '''
+        flutter:
+          config:
+            enable-flag: NOT-A-BOOLEAN
+        ''',
+      ),
+      throwsToolExit(message: 'must be a boolean'),
+    );
+  });
+
+  test('overrides from global configuration', () {
+    expect(
+      isEnabled(_configOnlyFeature, globalConfig: <String, Object>{'enable-flag': true}),
+      isTrue,
+    );
+  });
+
+  test('global configuration value must be a boolean', () {
+    expect(
+      () => isEnabled(
+        _configOnlyFeature,
+        globalConfig: <String, Object>{'enable-flag': 'NOT-A-BOOLEAN'},
+      ),
+      throwsToolExit(message: 'must be a boolean'),
+    );
+  });
+
+  test('overrides from local manifest take precedence over global configuration', () {
+    expect(
+      isEnabled(
+        _configOnlyFeature,
+        projectManifest: '''
+        flutter:
+          config:
+            enable-flag: true
+        ''',
+        globalConfig: <String, Object>{'enable-flag': false},
+      ),
+      isTrue,
+    );
+  });
+
+  test('overrides from environment', () {
+    expect(
+      isEnabled(_envOnlyFeature, environment: <String, String>{'ENABLE_FLAG': 'true'}),
+      isTrue,
+    );
+  });
+
+  test('overrides from environment are case insensitive', () {
+    expect(
+      isEnabled(_envOnlyFeature, environment: <String, String>{'ENABLE_FLAG': 'tRuE'}),
+      isTrue,
+    );
+  });
+
+  test('overrides from environment take precedence over configuration', () {
+    expect(
+      isEnabled(
+        _configAndEnvFeature,
+        environment: <String, String>{'ENABLE_FLAG': 'true'},
+        globalConfig: <String, Object>{'enable-flag': false},
+      ),
+      isTrue,
+    );
+  });
+}
+
+const Feature _noConfigFeature = Feature(name: 'example');
```

Comment:
Ubernit: should these constants be top of file for readability?

Reply:
Done.